src/io/obf.rs

```diff
@@ -170,20 +180,29 @@ pub async fn obfuscate<M, SU, SH, ST, R, P>(
         zeros.concat_rows(&[&identities])
     };
     log_mem("Computed u_0, u_1, u_star");
-
     #[cfg(feature = "debug")]
     handles.push(store_and_drop_matrix(b_star_cur.clone(), &dir_path, "b_star_0"));
 
     let mut pub_key_cur = pub_key_init;
 
+    /*
+    =============================================================================
+      Trapdoor preimage generation for input insertion step:
+
+      M_{i,b} ←$ B⁻¹_{i-1,*,γ_B} · ( U_b · B[i][b] )
+      N_{i,b} ←$ B⁻¹_{i,b,γ_B}   · ( U_* · B[i][*] )
+      K_{i,b} ←$ B⁻¹_{i,b,γ_B}   · (
+          -A_{att,i-1}·T_{att,i,b},
+           A_{att,i} - (0_{L_fhe+i,b,0_{L-i}} ⊗ G_{n+1}),
+           A_{t,i}
+        )
```

Comment:
I think there's some error in the definition of the target matrix for preimage K. I suggest to use a single encoding and a single public key in the comments (as we do in the codebase), without splitting between a_att and a_t